src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -72,18 +82,18 @@ public ProjectState(ProjectInfo projectInfo, HostLanguageServices languageServic
 
             var projectInfoFixed = FixProjectInfo(projectInfo);
 
-            _documentIds = projectInfoFixed.Documents.Select(d => d.Id).ToImmutableArray();
-            _additionalDocumentIds = projectInfoFixed.AdditionalDocuments.Select(d => d.Id).ToImmutableArray();
+            _documentIds = projectInfoFixed.Documents.Select(d => d.Id).ToImmutableList();
+            _additionalDocumentIds = projectInfoFixed.AdditionalDocuments.Select(d => d.Id).ToImmutableList();
 
             var parseOptions = projectInfoFixed.ParseOptions;
-            var docStates = ImmutableDictionary.CreateRange<DocumentId, DocumentState>(
+            var docStates = ImmutableSortedDictionary.CreateRange(
                 projectInfoFixed.Documents.Select(d =>
                     new KeyValuePair<DocumentId, DocumentState>(d.Id,
                         CreateDocument(d, parseOptions, languageServices, solutionServices))));
 
             _documentStates = docStates;
 
-            var additionalDocStates = ImmutableDictionary.CreateRange<DocumentId, TextDocumentState>(
+            var additionalDocStates = ImmutableSortedDictionary.CreateRange(DocumentIdComparer.Instance,
```

Comment:
He you specify the comparer, but you don't on line 89.

Reply:
Already fixed. Breaking a few thousand tests noticed it too.